ghost/core/test/e2e-api/admin/mail-events.test.js

```diff
@@ -0,0 +1,44 @@
+const {agentProvider} = require('../../utils/e2e-framework');
+const configUtils = require('../../utils/configUtils');
+
+describe('Mail Events API', function () {
+    let agent;
+
+    before(async function () {
+        agent = await agentProvider.getAdminAPIAgent();
+    });
+
+    it('Can add a mail event', async function () {
+        configUtils.set('hostSettings:siteId', 123);
+        configUtils.set('hostSettings:mailEventsSecretKey', 'foobarbaz');
+
+        const payload = {
+            // The signature is based on the previous two config values as well as the
+            // "events" array below. If you change any of these values, you will need to
+            // update the signature otherwise the request will fail
+            signature: 'c505cb9b343795954199d5d514bf520b6b133bd9f1580805f0fa150d1b89fdab',
+            events: [
+                {
+                    id: 'Ase7i2zsRYeDXztHGENqRA',
+                    timestamp: 1521243339.873676,
+                    event: 'opened',
+                    message: {
+                        headers: {
+                            'message-id': '20130503182626.18666.16540@sandboxb052085d6a7b401bb117d3a432d1d659.mailgun.org'
+                        }
+                    },
+                    recipient: 'alice@example.com'
+                }
+            ]
+        };
+
+        await agent
+            .post('/mail_events/', {
+                headers: {
+                    'content-type': 'application/json'
+                }
+            })
+            .body(payload)
```

Comment:
When using [e2e framework's assertions](https://www.notion.so/ghost/End-to-end-Testing-6a2ef073b1754b18aff42e24a632a007?pvs=4#b90c6286099849b493f6c7fbc11edd8e) it's important to add `matchBodySnapshot` and `matchHeaderSnapshot` in base tests like this one so we can have snapshots to track the response payload evolution over time. And catch bugs if the payload changes unintentionally.

Reply:
👍 Have added these assertions in